glance/api/v2/images.py

```diff
@@ -92,22 +93,51 @@ def __init__(self, db_api=None, policy_enforcer=None, notifier=None,
     def create(self, req, image, extra_properties, tags):
         image_factory = self.gateway.get_image_factory(req.context)
         image_repo = self.gateway.get_repo(req.context)
+
         try:
             if 'owner' not in image:
                 image['owner'] = req.context.project_id
 
-            api_policy.ImageAPIPolicy(req.context, image,
-                                      self.policy).add_image()
-
+            api_policy.ImageAPIPolicy(req.context, image, self.policy).add_image()
             ks_quota.enforce_image_count_total(req.context, req.context.owner)
-            image = image_factory.new_image(extra_properties=extra_properties,
-                                            tags=tags, **image)
+
+            # Attach domain info if enabled
+            if getattr(CONF, 'show_domain_info', False):
```

Comment:
Why would the config need a `getattr()`? We should have registered that option, so it should be always available - at least with its default value.